aws/resource_aws_eip_test.go

```diff
@@ -186,13 +186,17 @@ func TestAccAWSEIP_associated_user_private_ip(t *testing.T) {
 // Regression test for https://github.com/hashicorp/terraform/issues/3429 (now
 // https://github.com/terraform-providers/terraform-provider-aws/issues/42)
 func TestAccAWSEIP_classic_disassociate(t *testing.T) {
+	oldvar := os.Getenv("AWS_DEFAULT_REGION")
+	os.Setenv("AWS_DEFAULT_REGION", "us-east-1")
+	defer os.Setenv("AWS_DEFAULT_REGION", oldvar)
```

Comment:
🤔 is it also possible to define this via a provider block in the code?

Reply:
There's some historical context to this (of course there is 😅 ) you can grep and see this usage all over the testing code. We [default override this environment variable to `us-west-2`](https://github.com/terraform-providers/terraform-provider-aws/blob/c5824d5d74c22a048b5b1c576b70d86bead6ee20/aws/provider_test.go#L59-L85). I think the environment variable dance here is only really necessary for `ImportState: true` tests but we do it out of habit nowadays. At some point we need to tackle how to better handle multi-region/multi-credential testing. 🙁